packages/notifi-dapp-example/src/components/TopicStackRowInput.tsx

```diff
@@ -177,36 +206,67 @@ export const TopicStackRowInput: React.FC<TopicStackRowInputProps> = (
-                        );
-                      setFilterOptionsToBeSubscribed(
-                        updatedAlertFilterOptiopns,
-                      );
-                    },
-                  }}
-                />
-              );
-            },
-          )}
+          {isTopicGroup
+            ? correctUserInputParamsOrder(userInputParams).map(
+                (userInputParm, id) => {
+                  return (
+                    <TopicOptions<'group'>
+                      index={id}
+                      key={id}
+                      description={description}
+                      userInputParam={userInputParm}
+                      topics={props.topics}
+                      onSelectAction={{
+                        actionType: 'updateFilterOptions',
+                        action: (userInputParmName, option) => {
+                          if (!filterOptionsToBeSubscribed || !filterName)
+                            return;
+                          const updatedAlertFilterOptiopns =
+                            getUpdatedAlertFilterOptions(
+                              filterName,
+                              filterOptionsToBeSubscribed,
+                              userInputParmName,
+                              convertOptionValue(option, userInputParm.kind),
+                            );
+                          setFilterOptionsToBeSubscribed(
+                            updatedAlertFilterOptiopns,
+                          );
+                        },
+                      }}
+                    />
+                  );
+                },
+              )
+            : correctUserInputParamsOrder(userInputParams).map(
+                (userInputParm, id) => {
+                  return (
+                    <TopicOptions<'standalone'>
+                      index={id}
+                      key={id}
+                      description={description}
+                      userInputParam={userInputParm}
+                      topic={props.topic}
+                      onSelectAction={{
+                        actionType: 'updateFilterOptions',
+                        action: (userInputParmName, option) => {
+                          if (!filterOptionsToBeSubscribed || !filterName)
+                            return;
+                          const updatedAlertFilterOptiopns =
+                            getUpdatedAlertFilterOptions(
+                              filterName,
+                              filterOptionsToBeSubscribed,
+                              userInputParmName,
+                              convertOptionValue(option, userInputParm.kind),
+                            );
+                          setFilterOptionsToBeSubscribed(
+                            updatedAlertFilterOptiopns,
+                          );
+                        },
+                      }}
+                    />
+                  );
+                },
+              )}
```

Comment:
Just being curious, do we still need to ensure the correct order of userInputParms. Not sure if it will be confusing for user as the order is controlled by AP.

Reply:
Yes, I think we should for GMX as for the initial topics we haven't update the userInputParams with the correct order, but for new create topics, it should be the correct order, just incase we don't mess up the rendering for the  existing alerts